### PR TITLE
Added eslint support to phabricator

### PR DESCRIPTION
**Pull request by yangsu**
shameless copy pasta from https://secure.phabricator.com/D12198

@turadg ptal

CC @jnwng @eleith

### CODE REVIEW

**Comment by yangsu** [comment]
Once we start using this. i'll send this upstream to phabricator

File: src/lint/linter/ArcanistESLintLinter.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    protected function parseLinterOutput($path, $err, $stdout, $stderr) {
+        $lines = phutil_split_lines($stdout, false);
+
+        $messages = array();
+        foreach ($lines as $line) {
+            // Clean up nasty ESLint output
+            $clean_line = $output = preg_replace('!\s+!', ' ', $line);
+            $parts = explode(' ', ltrim($clean_line));
```
Comment (turadg) [inline src/lint/linter/ArcanistESLintLinter.php]: `parts[1]` gets repeated a lot. how about,
```
$clean_line = trim(preg_replace('!\s+!', ' ', $line));
list($severityStr, $lineNo) = explode(' ', $clean_line);

if ($severityStr == 'error' || $severityStr == 'warning') {
  $severity = $severityStr === 'error' ?
…
   $message->setLine($lineNo);
…
  $message->setDescription($clean_line);
…
```

File: src/lint/linter/ArcanistESLintLinter.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            if (isset($parts[1]) && ($parts[1] === 'error' || $parts[1] === 'warning')) {
+                $severity = $parts[1] === 'error' ?
+                    ArcanistLintSeverity::SEVERITY_ERROR :
+                    ArcanistLintSeverity::SEVERITY_WARNING;
+
+                $message = new ArcanistLintMessage();
+                $message->setPath($path);
+                $message->setLine($parts[0]);
+                $message->setCode($this->getLinterName());
+                $message->setDescription(implode(' ', $parts));
```
Comment (turadg) [inline src/lint/linter/ArcanistESLintLinter.php]: not sure why this is imploded again?
```
…
  $message->setDescription($clean_line);
…

**Comment by turadg** [comment]
gtg

**Pull request merged**